### ISSUE TITLE
Aliasing generic type containing typevar doesn't work

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1339]
**Describe the bug**

```python3
from typing import List, TypeVar

T = TypeVar("T")
L = List[T]

def f(x: L) -> L:
    ...

f([1, 2, 3]) # Error
```

Outputs:
```
  11:3 - error: Argument of type "list[int]" cannot be assigned to parameter "x" of type "L[T@L]" in function "f"
    TypeVar "_T@list" is invariant
      Type "int" cannot be assigned to type "T@L" (reportGeneralTypeIssues)
```

It works if the alias `L` is not used:

```python3
from typing import List, TypeVar

T = TypeVar("T")


def f(x: List[T]) -> List[T]: # not use alias `L`
    ...


f([1, 2, 3]) # OK
```

This problem also occurs for covariant container type:

```python3
from typing import List, TypeVar, FrozenSet

T = TypeVar("T")
L = FrozenSet[T] # Use FrozenSet (covariant type) insead of List

def f(x: L) -> L:
    ...
f(frozenset([1, 2, 3]))  # Error
```

Mypy doesn't output any error.

**Expected behavior**

No error

**Additional context**

pyright: 1.1.99

**Comment by erictraut** [comment]
Pyright is doing the right thing here. You're defining a generic type alias `L` that takes a single type argument, but then you're failing to provide a type argument for it when you use it. 

The following is a correct usage of a generic type alias `L`:

```python
def f(x: L[T]) -> L[T]:
    ...
```

Arguably, Pyright should provide an error when you attempt to use `L` without any type arguments (since it is defined such that it requires a single type argument). I'll look into adding that error.

**Comment by erictraut** [comment]
There was already code in place that was supposed to emit an error when a generic type alias was used without any type arguments provided, but there was a bug in this code. I've fixed that bug. Here's how it looks with that fix in place:

<img width="706" alt="Screen Shot 2021-01-06 at 7 34 46 AM" src="https://user-images.githubusercontent.com/7040122/103786988-e99c9d80-4ff1-11eb-891f-e2851978503b.png">

**Comment by tamuhey** [comment]
Thanks!!

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.100, which I just published. It will also be included in the next release of Pylance.
